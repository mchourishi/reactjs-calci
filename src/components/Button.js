import React from "react";
import PropTypes from "prop-types";

class Button extends React.Component {
  handleClick = () => {
    this.props.clickHandler(this.props.name);
  };
  render() {
    const className = ["calculator-key", this.props.cls ? this.props.cls : ""];

    return (
      <div>
        <button onClick= {this.handleClick} className={className.join(" ").trim()}>
          {this.props.name}
        </button>
      </div>
    );
  }
}

Button.propTypes = {
  name: PropTypes.string,
  cls: PropTypes.string,
  clickHandler: PropTypes.func
};
export default Button;
