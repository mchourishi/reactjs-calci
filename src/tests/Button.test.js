import React from "react";
import { shallow } from "enzyme";
import Button from "../components/Button";

describe("Button Component", () => {
  let wrapper;
  beforeEach(() => (wrapper = shallow(<Button />)));

  it("should render correctly", () => expect(wrapper).toMatchSnapshot());

  it("should render the button", () => {
    expect(wrapper.containsMatchingElement(<button />)).toEqual(true);
  });

  it("Test click event", () => {
    const mockCallBack = jest.fn();
    const button = shallow(<button onClick={mockCallBack}>Ok!</button>);
    button.find("button").simulate("click");
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });

  it("button should have class calculator-key", () => {
    expect(wrapper.find("button").hasClass("calculator-key")).toEqual(true);
  });
});
