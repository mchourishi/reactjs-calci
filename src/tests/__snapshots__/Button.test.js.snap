// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        className="calculator-key"
        onClick={[Function]}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "calculator-key",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          className="calculator-key"
          onClick={[Function]}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "calculator-key",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
